recipes-app/iot2050-firmware-update/files/iot2050-firmware-update

```diff
@@ -43,16 +72,47 @@ class FirmwareUpdate(object):
             FirmwareUpdateException("sha256 file not exists")
 
         try:
-            self.firmware.seek(0)
+            self.firmware_tar.seek(0)
             method = hashlib.sha256()
-            method.update(self.firmware.read())
+            method.update(self.firmware_tar.read())
             hash_code = method.hexdigest()
         except IOError as e:
-            print("Reading {} failed: {}".format(self.firmware.name, e.strerror))
+            print("Reading {} failed: {}".format(self.firmware_tar.name, e.strerror))
             sys.exit(1)
 
         if file_sha256 != hash_code:
-            FirmwareUpdateException("Image checksum is not correct")
+            FirmwareUpdateException(
+                "Tarball {} checksum is not correct".format(self.firmware_tar.name))
+
+    def verify_tar(self):
```

Comment:
This seems to miss a force mode.

Reply:
I was not sure if it's a good idea to provide that function... 

However, without this, for some advanced users, it's might be not so handy to use this tool to flash a single firmware bin, provided they know what they are doing.

OK, I think it's good to do that, however before the effective opration on the flash, we need ask the user that they must know what they are doing.

Reply:
Force it required, already for development purposes, but also for cases that we may have missed.